youtube_dl/extractor/bandcamp.py

```diff
@@ -248,3 +241,78 @@ def _real_extract(self, url):
             'title': title,
             'entries': entries,
         }
+
+
+class BandcampArtistDefaultAlbumIE(BandcampAlbumIE):
```

Comment:
```python
$ python2
> album_VALID_URL = r'https?://(?:(?P<subdomain>[^.]+)\.)?bandcamp\.com/album/(?:(?P<album_id>[^?#]+)|/?(?:$|[?#]))'
> artist_default_album_VALID_URL = r'https?://(?P<subdomain>[^.]+)\.bandcamp\.com/?(?:$|[?#])(?P<album_id>)'
> bool(re.match(album_VALID_URL, 'http://dotscale.bandcamp.com'))
False
> bool(re.match(artist_default_album_VALID_URL, 'http://dotscale.bandcamp.com'))
True
```

The extractor for the album extractor had to be made less general so it did not clobber the Bandcamp Weekly extractor.  However, the resulting regex would not capture cases such as 'http://dotscale.bandcamp.com' where an album is displayed on an "artists" page.  Thus this extractor to preserve capturing this case.